src/main/java/me/dinowernli/grpc/polyglot/config/CommandLineArgs.java

```diff
@@ -16,107 +16,95 @@
-
-  @Option(name = "--add_protoc_includes", metaVar = "<path1>,<path2>")
+  @Parameter(names = "--add_protoc_includes", description ="<path1>,<path2>", order = 0)
   private String addProtocIncludesArg;
-
-  @Option(name = "--proto_discovery_root", metaVar = "<path>")
+  @Parameter(names = "--proto_discovery_root", description ="<path>", order = 0)
   private String protoDiscoveryRootArg;
-
-  @Option(name = "--deadline_ms", metaVar = "<number>")
-  private Integer deadlineMs;
-
-  @Option(name = "--tls_ca_cert_path", metaVar = "<path>")
-  private String tlsCaCertPath;
-
-  @Option(name = "--tls_client_cert_path", metaVar = "<path>")
-  private String tlsClientCertPath;
-
-  @Option(name = "--tls_client_key_path", metaVar = "<path>")
-  private String tlsClientKeyPath;
-
-  @Option(name = "--tls_client_override_authority", metaVar = "<host>")
-  private String tlsClientOverrideAuthority;
-
-  @Option(name = "--use_reflection", metaVar = "true|false")
+  @Parameter(names = "--use_reflection", description ="true|false", order = 0)
   private String useReflection;
+ @Parameter(names = "--help", help = true, order = 1)
+  private boolean help;
 
-  @Option(name = "--help")
-  private Boolean help;
-
-  // *************************************************************************
-  // * Initial step towards the migration to "polyglot <command> [flagz...]" *
-  // *************************************************************************
+  // Commands
 
   /** Command to make a GRPC call to an endpoint */
   public static final String CALL_COMMAND = "call";
-
   /** Command to list all known services defined in the proto files*/
   public static final String LIST_SERVICES_COMMAND = "list_services";
-
-  @Option(name = "--command", metaVar = "<call|list_services>")
+  /** Captures the command called */
   private String commandArg;
 
-  // TODO: Move to a "list_services"-specific flag container
-  @Option(
-      name = "--service_filter",
-      metaVar = "service_name",
-      usage="Filters service names containing this string e.g. --service_filter TestService")
-  private String serviceFilterArg;
-
-  // TODO: Move to a "list_services"-specific flag container
-  @Option(
-      name = "--method_filter",
-      metaVar = "method_name",
-      usage="Filters service methods to those containing this string e.g. --method_name List")
-  private String methodFilterArg;
-
-  //TODO: Move to a "list_services"-specific flag container
-  @Option(
-      name = "--with_message",
-      metaVar = "true|false",
-      usage="If true, then the message specification for the method is rendered")
-  private String withMessageArg;
-
-  @Option(name = "--metadata", metaVar = "k1:v1,k2:v2,...")
-  private String metadataArg;
+  private final CallCommand callCommand = new CallCommand();
+  private final ListServicesCommand listServicesCommand = new ListServicesCommand();
+
+  @Parameters(separators = "=", commandDescription = "Make a GRPC call to an endpoint")
```

Comment:
Same here, should we support spaces? (and in ListCommand below)

Reply:
Sure, done it.